trieste/acquisition/rule.py

```diff
@@ -94,27 +88,39 @@ def __init__(
         self,
         builder: AcquisitionFunctionBuilder | None = None,
         optimizer: AcquisitionOptimizer[SP_contra] | None = None,
+        num_query_points: int = 1,
     ):
         """
         :param builder: The acquisition function builder to use. Defaults to
-            :class:`~trieste.acquisition.ExpectedImprovement` with tag :data:`OBJECTIVE`.
+                :class:`~trieste.acquisition.ExpectedImprovement` with tag :data:`OBJECTIVE`.
         :param optimizer: The optimizer with which to optimize the acquisition function built by
-            ``builder``. This should *maximize* the acquisition function, and must be compatible
-            with the global search space. Defaults to
-            :func:`~trieste.acquisition.optimizer.optimize`.
+                ``builder``. This should *maximize* the acquisition function, and must be compatible
+                with the global search space. Defaults to
+                :func:`~trieste.acquisition.optimizer.optimize`.
+        :param num_query_points: The number of points to acquire.
         """
+
+        if not num_query_points > 0:
+            raise ValueError(
+                f"Number of query points must be greater than 0, got {num_query_points}"
+            )
```

Comment:
i'm starting to prefer `tf.debugging` where possible cos it becomes part of the graph when compiled. It also comes with free error messages. What say you?

Reply:
yeah for sure - they're everywhere